recipes/compute-pascals-triangle.md

```diff
@@ -0,0 +1,49 @@
+# Compute Pascal's triangle
+
+## Problem
+
+Display the top `n` rows of
+[Pascal's triangle](https://en.wikipedia.org/wiki/Pascal's_triangle).
+
+## Solution
+
+```Scheme
+(define (pascal n k)
+  (cond ((or (= k 0) (= k n))
+         1)
+        ((< 0 k n)
+         (+ (pascal (- n 1) (- k 1))
+            (pascal (- n 1) k)))
+        (else
+         (error "Bad indexes" n k))))
+
+(define (display-triangle rows)
+  (let loop-rows ((n 0))
+    (when (< n rows)
+      (let loop-cols ((k 0))
+        (cond ((<= k n)
+               (display (pascal n k))
+               (display #\space)
+               (loop-cols (+ k 1)))
+              (else
+               (newline)
+               (loop-rows (+ n 1))))))))
```

Comment:
Maybe we should name this just `loop` to keep consistency, in the matrix I've changed row/cols into inner/outer.

Reply:
OK, I'll rename to inner/outer.